ewatercycle/models/abstract.py

```diff
@@ -43,7 +46,7 @@ def initialize(self, config_file: str) -> None:
     def finalize(self) -> None:
         """Perform tear-down tasks for the model."""
         self.bmi.finalize()
-        # TODO terminate container if running?
+        self.bmi = None
```

Comment:
```suggestion
        self.__del__()
```

Reply:
That wouldn't work, as `__del__()` deletes the entire `model` instance, not just the `bmi` attribute.

Reply:
Okay, let me rephrase that. 
- `model.finalize() is a BMI function to wrap up the model run, and as a side effect we want to tear down the container. As a result of this operation, `model.bmi` should be destroyed, but `model` itself will still exist.
- `model.__del__()` is called before the *model instance* is finalized, so e.g. when a user calls `del model` and there's no other remaining references to `model`, so it will be garbage collected. In that case, we want to explicitly ensure that the container is destroyed as well.

Although you could technically call `self.__del__()` from within `model.finalize()`, I think this is not what the `_del__()` method is intended for, and I'd rather not do that.